dartsim/src/plugin.cc

```diff
@@ -62,6 +62,7 @@ class Plugin :
     public virtual SimulationFeatures,
     public virtual WorldFeatures { };
 
+#ifndef GZ_PHYSICS_BAZEL_BUILD
```

Comment:
FYI @azeey this is what breaks when dartsim is statically linked via bazel.  Removing this section causes the cashes to go away.

Reply:
💸

Reply:
I found that loading the plugins with `RTLD_NODELETE` and removing this section fixes the issue on macOS and Linux. We do that for Gazebo plugins already, so I think we should do the same for physics engines.